pages/index.tsx

```diff
@@ -25,11 +26,42 @@ export default function Home() {
   }, [visibleDistrictIndices]);
 
   const [error, setError] = useState('');
+  const [hint, setHint] = useState('');
 
   const inputElementRef = useRef<HTMLInputElement>(null);
 
+  const addDistrictFromHint = (hint: string) => {
+    if (hint != '') {
+      setHint('');
+    }
+    if (error != '') {
+      setError('');
+    }
+
+    const userInput = hint;
+    const districtIndex = DistrictData.findIndex(
+      (d) => d.name.toLowerCase() === userInput.toLowerCase()
+    );
+    if (visibleDistrictIndices.includes(districtIndex)) {
+      setError(
+        `${DistrictData[districtIndex].name} has already been submitted.`
+      );
+    } else if (districtIndex >= 0 && inputElementRef.current?.value) {
+      inputElementRef.current.value = '';
+      setVisibleDistrictIndices([...visibleDistrictIndices, districtIndex]);
+    } else {
+      setError(`No district named ${userInput}`);
+    }
+  };
```

Comment:
addDistrictFromHint and addDistrict are very similar maybe you can create a more general function? Just some thoughts.

Reply:
Yup, tried doing that but since the two functions take two different types of parameters the code started getting messy. So decided to addDistrictFromHint implementation in a different function.